crates/nu-cli/src/data/value.rs

```diff
@@ -159,13 +158,50 @@ pub fn format_leaf<'a>(value: impl Into<&'a UntaggedValue>) -> DebugDocBuilder {
     InlineShape::from_value(value.into()).format().pretty()
 }
 
-pub fn style_leaf<'a>(value: impl Into<&'a UntaggedValue>) -> &'static str {
-    match value.into() {
-        UntaggedValue::Primitive(p) => style_primitive(p),
-        _ => "",
+pub fn style_leaf(value: &Value) -> String {
+    match value {
+        Value {
+            value: UntaggedValue::Primitive(p),
+            tag,
+        } => {
+            let mut styled_primitive_string = style_primitive(&p).to_string();
+
+            if let Some(data_color) = tag.data_color {
+                if let Some(data_color_string) = color_to_str(data_color) {
+                    styled_primitive_string =
+                        format!("F{}{}", data_color_string, style_primitive(&p));
+                }
+            }
+
+            styled_primitive_string
+        }
+        _ => "".to_string(),
     }
 }
 
+fn color_to_str(color: color::Color) -> Option<String> {
+    let color_string = match color {
+        color::GREEN => "g",
+        color::RED => "r",
+        color::BLUE => "u",
+        color::BLACK => "b",
+        color::YELLOW => "y",
+        color::MAGENTA => "m",
+        color::CYAN => "c",
+        color::WHITE => "w",
+        color::BRIGHT_GREEN => "bg",
+        color::BRIGHT_RED => "br",
+        color::BRIGHT_BLUE => "bu",
+        color::BRIGHT_YELLOW => "by",
+        color::BRIGHT_MAGENTA => "bm",
+        color::BRIGHT_CYAN => "bc",
+        color::BRIGHT_WHITE => "bw",
+        _ => return None,
+    };
+
+    Some(color_string.to_string())
+}
+
```

Comment:
It is kind of weird making a reverse map of `str_to_color()`, but I wanted a user to be able to choose a color for their data by using the enum (e.g. `color::GREEN`) over having to figure out how to format the styling string themselves.  So they just pick a color using the color enum and the string is built automatically, which mitigates mistakes (hopefully).